### PR TITLE
Validate calls to ebpf_map_update_elem()

### LINKED ISSUE #170: Exposing kernel pointers to user mode apps
The following program passes the verifier:
```
int func(struct ctx* ctx)
{
    uint32_t key = 0;
    return bpf_map_update_elem(&map, &key, &ctx, 0);
}
```

As a result, the ctx pointer (or a pointer into the stack, etc.) can be stored in a map.
Issue is that can expose kernel addressing details to user mode apps.
The verifier does fail code that tries to use a value returned from bpf_map_lookup_elem as a pointer, so that part is fine, but it also means there's little legitimate reason to store a pointer in a map value since once there it's not a  safe pointer.

Is there any reason this should be allowed?  Seems like the verifier should fail code like above.

### PR DESCRIPTION
**Pull request by elazarg**
If the a lower bound on the start address is known, iterate through all bytes and check in the bitmap to see they hold numerical values. Only check for stack, since packet is guaranteed to hold only numerical values.

Fix #170.

Signed-off-by: Elazar Gershuni <elazarg@gmail.com>

### CODE REVIEW

**Comment by elazarg** [comment]
@dthaler what do you think?

**Comment by caballa** [comment]
It looks good to me.

File: src/crab/bitset_domain.hpp [hunk 1]
```diff
@@ -76,4 +76,15 @@ class bitset_domain_t final {
     }
 
     friend std::ostream& operator<<(std::ostream& o, const bitset_domain_t& array);
+
+    bool all_num(int lb, int ub) {
```
Comment (dthaler) [inline src/crab/bitset_domain.hpp:80]: Assuming the lower bound is inclusive and the upper bound is exclusive, the code looks right, but a comment to the effect would help.
```suggestion
    // Test whether all values in the range [lb,ub) are numerical.
    bool all_num(int lb, int ub) {
```

**Review by dthaler — APPROVED** [review]

**Comment by dthaler** [comment]
I'd recommend merging #171, then in this PR update test_verify.cpp to reverse the test to expect fail, so that the CI/CD test pass on this PR actually shows it fail.

File: src/test/test_verify.cpp [hunk 2]
```diff
@@ -315,9 +323,8 @@ TEST_SECTION("suricata", "xdp_filter.o", "xdp")
 
 // Test some programs that ought to fail verification.
 //
-// False negative: exposing kernel pointers to user-mode apps
-// This should be changed to TEST_SECTION_FAIL once fixed.
```
Comment (elazarg) [inline src/test/test_verify.cpp:319]: `TEST_SECTION_FAIL` is for false positives; added `TEST_SECTION_REJECT` macro for negatives.

**Pull request merged**